File: 998-check-completeness-of-a-binary-tree/check-completeness-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int calculate(TreeNode*root)
    {
        if(!root)
        {
            return 0;
        }
        return 1+calculate(root->left)+calculate(root->right);
    }
    bool check(TreeNode*root,int TotalNode,int i)
    {
        if(!root)
        {
            return true;
        }
        if(i>TotalNode)
        {
            return false;
        }
        return check(root->left,TotalNode,2*i) && check(root->right,TotalNode,2*i+1);
    }
    bool isCompleteTree(TreeNode* root) {
        int TotalNode=calculate(root);
         return check(root,TotalNode,1);
    }
};

File: 998-check-completeness-of-a-binary-tree/check-completeness-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public static int find(TreeNode root)
    {
        if(root==null)
        {
            return 0;
        }
        return 1+find(root.left)+find(root.right);
    }
    public static boolean check(TreeNode root,int i,int totalNode)
    {
        if(root==null)
        {
            return true;
        }
        if(i>totalNode)
        {
            return false;
        }
        return check(root.left,2*i,totalNode) && check(root.right,2*i+1,totalNode);
    }
    public boolean isCompleteTree(TreeNode root) {
        int totalNode=find(root);
        return check(root,1,totalNode);
    }
}